opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -100,17 +106,72 @@ message MetricDescriptor {
   repeated string label_keys = 5;
 }
 
-// A collection of data points that describes the time-varying values
-// of a metric.
-message TimeSeries {
+// Int64TimeSeriesList is a list of timeseries of int64 values.
+message Int64TimeSeriesList {
```

Comment:
Can we get rid of the "List" messages? and have oneof { repeated xyz }?

Reply:
No, unfortunately that's not valid in Protobuf :-(

Reply:
But we can get rid of the entire oneof then. The type of the data is known from the metric descriptor. If [gauge|couter]_long then is Int64, etc.

Reply:
I am not sure I understand how your propose the message structure to be. Do you mean get rid of oneof and just list all fields (flatten as we did for Span AttributeValue)? Like this:
```protobuf
// Defines a Metric which has one or more timeseries.
message Metric {
  // The descriptor of the Metric.
  MetricDescriptor metric_descriptor = 1;

  // The resource for the metric. If unset, it may be set to a default value
  // provided for a sequence of messages in an RPC stream.
  opentelemetry.proto.resource.v1.Resource resource = 2;

  // data is a list of one or more TimeSeries for a single metric, where each timeseries has
  // one or more points. Only one of the following fields is used for the data, depending on
  // the type of the metric defined by MetricDescriptor.type field.
  repeated Int64TimeSeries int64_data = 3;
  repeated DoubleTimeSeries double_data = 4;
  repeated HistogramTimeSeries histogram_data = 5;
  repeated SummaryTimeSeries summary_data = 6;
}
```
This will work.

Reply:
Yes that is what I think will be better. Not sure if the usage of keyward data is right but the structure looks like what I thought

Reply:
Yeah, I think this is even better (and will use less CPU). It will mean very small increase of in-memory size but simplies the message structure and removes a lot of message definitions. Let me change.

Reply:
Done.